packages/back-end/src/models/OrganizationModel.ts

```diff
@@ -434,3 +436,98 @@ export async function updateMember(
     }),
   });
 }
+
+/**
+ * Delete an organization and all associated data
+ */
+export async function deleteOrganizationData(orgId: string) {
+  logger.info("Deleting org %s", orgId);
+
+  const org = await getOrganizationById(orgId);
+
+  if (!org) throw new Error("Organization not found");
+
+  const userIds = org.members.map((m: Member) => m.id);
+
+  // TODO better way to do this?
+  const usersToDelete = [];
+  for (const userId of userIds) {
+    const allOrgsWithUser = await findOrganizationsByMemberId(userId);
+    if (allOrgsWithUser.length === 1) usersToDelete.push(userId);
+  }
```

Comment:
Great catch, thanks.

RE: The sub-division use case; to limit the scope of this PR, I think it would be reasonable to ask orgs to create new members before deleting an old org to prevent losing those users. What do you think?

Reply:
To be honest I'm not entirely sure what is actually lost when a user is permanently deleted and then re-added, especially considering that any comments or other such things that were part of the organization they were part of just got deleted.  I suppose they might have links to discussions in other orgs that they were once a part of but are no more.  However the user info seems denormalized on discussion, anyway.  So it probably doesn't matter.  Maybe worth asking @jdorn in case there is something I'm forgetting.

Reply:
I don't have complete context, but I think we should not delete users from Mongo and instead rely on the Orphaned Users UI.  We can improve that component to better handle bulk actions if that's a concern.